invenio_records_marc21/records/systemfields/providers.py

```diff
@@ -32,3 +32,28 @@ class MarcDraftProvider(MarcRecordProvider):
     """
 
     default_status_with_obj = PIDStatus.NEW
+
+    predefined_pid_value = ""
+
+    @classmethod
+    def create(cls, object_type=None, object_uuid=None, options=None, **kwargs):
+        """Intermediate step to create the marcid.
+
+        With this intermediate step it is possible to set pre-calculated marcid's.
+        """
+        if (
+            "record" in kwargs and "$schema" in kwargs["record"]
```

Comment:
does this also checks if the `predefined_pid_value` already exist in the system?

Reply:
if i remember correctly, then the super call should do that